blocks/api/post.pegjs

```diff
@@ -1,8 +1,14 @@
 {
 
+function untransformValue( value ) {
+	return 'string' === typeof value
+		? value.replace( /\\-/g, '-' )
```

Comment:
this needs to pair with the serializer to make sure we don't get here. we have _have_ to face escaping in some manner. doing it here I think is the easiest way to accomplish this: "You _must_ escape hyphens. You need not escape _anything else_."